### ISSUE TITLE
Java8 and CallerSensitive annotation error

### ISSUE THREAD

**Issue by fixundfertig123** [anvie/blueprints-scala#32]
Hey,

I am simply not sure whether this is a bug, or whether I am the problem, so sorry for that.

Some facts:
- Java 1.8.0_40
- Scala 2.11.5
- sbt 0.13.7

When runing a simple scala main class (OrientDBMigrator) I get an exception thrown when reaching the "save" part.

	val address = Jakarta()
	address.province = "Jakarta Barat" // this will be saved
	address.street = "KS. Tubun" // this will be saved
	address.postalCode = 12345 // this will be saved
	address.save() ---> Exception thrown

[error] Exception in thread "main" java.lang.InternalError: CallerSensitive annotation expected at frame 1
[error]         at sun.reflect.Reflection.getCallerClass(Native Method)
[error]         at com.ansvia.graph.BlueprintsWrapper$.<init>(BlueprintsWrapper.scala:21)
[error]         at com.ansvia.graph.BlueprintsWrapper$.<clinit>(BlueprintsWrapper.scala)
[error]         at com.ansvia.graph.BlueprintsWrapper$DbObject$class.save(BlueprintsWrapper.scala:306)
[error]         at Jakarta.save(OrientDBMigrator.scala:41)
[error]         at OrientDBMigrator$.main(OrientDBMigrator.scala:74)
[error]         at OrientDBMigrator.main(OrientDBMigrator.scala)

I am thankful for any comment,

Cheers, Fritz

**Comment by AlexanderRay** [comment]
+1

**Comment by fixundfertig123** [comment]
Hey 

@anvie @pdolega: I looked through the code and saw that merge https://github.com/anvie/blueprints-scala/commit/1fdaca2ec1ad30f22cafda793244c33972be098c causes the problem. I just removed this merge and now the test class work fine but one. See below:

<code>
[info] x using mutual wrapper #2
[error]    'neptune' is not equal to 'jupiter' (BlueprintsWrapperSpec.scala:103)
[info]
</code>

However,  undoing solves my issue of CallerSensitive annotation error but brings back the former problem of issue https://github.com/anvie/blueprints-scala/pull/28.

Cheers, Fritz

**Comment by fixundfertig123** [comment]
Not knowing too much about reflection this looks to me relevant:
https://issues.scala-lang.org/browse/SI-6789

**Comment by AlexanderRay** [comment]
``` scala
val defaultClassloader = if(Reflection.getCallerClass!=null) Reflection.getCallerClass.getClassLoader else null

```

causes that problem. may be it should be a "lazy val" so I can avoid to call of "Reflection.getCallerClass" and set the ClassLoader by "classOf[MyClass].getClassLoader"

**Comment by pdolega** [comment]
@fixundfertig123 your environment setup is close to what I have right now so I should be able to reproduce it easily. I'll investigate this issue today EOD and let you know (hopefully bringing also fix)

**Comment by pdolega** [comment]
Hey guys. Got little delay but issue is fixed here: https://github.com/anvie/blueprints-scala/pull/33
Also added builds for Java 8 so things like that get tested automatically.

@anvie take a look whenever you have some spare time.

**Comment by fixundfertig123** [comment]
@anvie, @pdolega thanks for your work!!!

**Comment by mikolak-net** [comment]
@anvie it seems there is a regression in the 1.50 release (the most recent one in Maven Central). It looks like the fix never got through to the 2.11 branch. Could this be fixed?
